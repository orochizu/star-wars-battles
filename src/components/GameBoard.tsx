import React, { useEffect, useState } from 'react';

import { useTranslation } from 'react-i18next';

import { CircularProgress, Grid, Typography } from '@material-ui/core';

import { Card } from '../types';
import { RoundStatus } from '../types';

import PreviewCard from './PreviewCard';

const defaultCard: Card = {
    title: 'NO_CARD_SELECTED',
    status: RoundStatus.WAITING,
};

const GameBoard: React.FC<GameBoardProps> = ({
    loading,
    cardOne = defaultCard,
    cardTwo = defaultCard,
}: GameBoardProps) => {
    const { t } = useTranslation();

    const [[playerOnePoints, playerTwoPoints], setPoints] = useState<
        [number, number]
    >([0, 0]);

    useEffect(() => {
        setPoints([
            cardOne.status === RoundStatus.WIN
                ? playerOnePoints + 1
                : playerOnePoints,
            cardTwo.status === RoundStatus.WIN
                ? playerTwoPoints + 1
                : playerTwoPoints,
        ]);
        // There is no need to run effect when points are changed
        // eslint-disable-next-line
    }, [cardOne, cardTwo]);

    return (
        <>
            {loading ? (
                <>
                    <Grid item xs="auto">
                        <CircularProgress size={60} />
                    </Grid>
                    <Grid item xs={12}>
                        <Typography align="center">{t('LOADING')}</Typography>
                    </Grid>
                </>
            ) : (
                <>
                    <Grid item xs={6}>
                        <PreviewCard
                            title={t(cardOne.title)}
                            variant="resistance"
                        >
                            <Grid container justify="center">
                                <Grid item xs="auto">
                                    <Typography variant="h4">
                                        {playerOnePoints}
                                    </Typography>
                                </Grid>
                            </Grid>
                        </PreviewCard>
                    </Grid>
                    <Grid item xs={6}>
                        <PreviewCard title={t(cardTwo.title)} variant="empire">
                            <Grid container justify="center">
                                <Grid item xs="auto">
                                    <Typography variant="h4">
                                        {playerTwoPoints}
                                    </Typography>
                                </Grid>
                            </Grid>
                        </PreviewCard>
                    </Grid>
                </>
            )}
        </>
    );
};

interface GameBoardProps {
    loading: boolean;
    cardOne?: Card;
    cardTwo?: Card;
}

export default GameBoard;
