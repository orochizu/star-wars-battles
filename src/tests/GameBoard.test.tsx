import React from 'react';

import { createRenderer } from 'react-test-renderer/shallow';

import GameBoard from '../components/GameBoard';

jest.mock('react-i18next', () => {
    return {
        useTranslation: () => ({
            t: (key: string) => key.toUpperCase(),
        }),
    };
});

const component = <GameBoard loading={false} />;

it('GameBoard.tsx should match snapshot', () => {
    const renderer = createRenderer();
    renderer.render(component);

    const snapshot = renderer.getRenderOutput();

    expect(snapshot).toMatchSnapshot();
});
