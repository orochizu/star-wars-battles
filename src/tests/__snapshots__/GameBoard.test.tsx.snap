// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard.tsx should match snapshot 1`] = `
<React.Fragment>
  <React.Fragment>
    <ForwardRef(WithStyles)
      item={true}
      xs={6}
    >
      <ForwardRef(WithTheme)
        title="NO_CARD_SELECTED"
        variant="resistance"
      >
        <ForwardRef(WithStyles)
          container={true}
          justify="center"
        >
          <ForwardRef(WithStyles)
            item={true}
            xs="auto"
          >
            <ForwardRef(WithStyles)
              variant="h4"
            >
              0
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithTheme)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      xs={6}
    >
      <ForwardRef(WithTheme)
        title="NO_CARD_SELECTED"
        variant="empire"
      >
        <ForwardRef(WithStyles)
          container={true}
          justify="center"
        >
          <ForwardRef(WithStyles)
            item={true}
            xs="auto"
          >
            <ForwardRef(WithStyles)
              variant="h4"
            >
              0
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithTheme)>
    </ForwardRef(WithStyles)>
  </React.Fragment>
</React.Fragment>
`;
